core/app/models/spree/promotion_code_batch.rb

```diff
@@ -0,0 +1,31 @@
+module Spree
+  class PromotionCodeBatch < ActiveRecord::Base
+    belongs_to :promotion, class_name: "Spree::Promotion"
+    has_many :promotion_codes, class_name: "Spree::PromotionCode", dependent: :destroy
+
+    validates :number_of_codes, numericality: { greater_than: 0 }
+    validates_presence_of :base_code, :number_of_codes
+
+    def status
+      if error.present?
```

Comment:
Can you return some kind of status here instead? something with a symbol or something. so we can use locales to display the actual text. When we hardcode english like this, people that speak weird languages become sad and complain to me in private messages.